sys/include/ztimer64.h

```diff
@@ -0,0 +1,519 @@
+ *
+ * Here's a table for absolute variants:
+ *
+ * | relative version               | absolute version               |
+ * |--------------------------------|--------------------------------|
+ * | ztimer64_set()                 | ztimer64_set_at()              |
+ * | ztimer64_spin()                | ztimer64_spin_until()          |
+ * | ztimer64_sleep()               | ztimer64_sleep_until()         |
+ * | ztimer64_set_msg()             | ztimer64_set_msg_at()          |
+ * | ztimer64_set_wakeup()          | ztimer64_set_wakeup_at()       |
+ * | ztimer64_set_timeout_flag()    | ztimer64_set_timeout_flag_at() |
+ * | ztimer64_msg_receive_timeout() | ztimer64_msg_receive_until()   |
+ * | ztimer64_mutex_lock_timeout()  | ztimer64_mutex_lock_until()    |
+ * | ztimer64_rmutex_lock_timeout() | ztimer64_rmutex_lock_until()   |
+ *
+ *
+ * @{
+ *
+ * @file
+ * @brief       ztimer 64bit API
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef ZTIMER64_H
+#define ZTIMER64_H
+
+#include <stdint.h>
+
+#include "mutex.h"
+#include "msg.h"
+#include "ztimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ztimer64_t forward declaration
+ */
+typedef struct ztimer64_base ztimer64_base_t;
+/**
+ * @brief ztimer64_clock_t forward declaration
+ */
+typedef struct ztimer64_clock ztimer64_clock_t;
+
+/**
+ * @brief   Minimum information for each timer
+ */
+struct ztimer64_base {
+    ztimer64_base_t *next;          /**< next timer in list */
+    uint64_t target;                /**< absolute target time */
+};
+
+/**
+ * @brief   ztimer64 structure
+ *
+ * This type represents an instance of a timer, which is set on an
+ * underlying clock object
+ */
+typedef struct {
+    ztimer64_base_t base;               /**< clock list entry */
+    void (*callback)(void *arg);        /**< timer callback function pointer */
+    void *arg;                          /**< timer callback argument */
+} ztimer64_t;
+
+/**
+ * @brief   ztimer64 clock structure
+ */
+struct ztimer64_clock {
+    ztimer64_base_t *first;         /**< list of active timers              */
+    ztimer_clock_t *base_clock;     /**< 32bit clock backend                */
+    ztimer_t base_timer;            /**< 32bit backend timer                */
+    uint32_t base_last;             /**< lower timer last checkpoint time   */
+    uint64_t checkpoint;            /**< lower timer checkpoint offset      */
+    uint16_t adjust_set;            /**< will be subtracted on every set()  */
+    uint16_t adjust_sleep;          /**< will be subtracted on every sleep(),
+                                         in addition to adjust_set          */
+#if MODULE_PM_LAYERED || DOXYGEN
+    uint8_t block_pm_mode;          /**< min. pm mode to block for the clock to run */
+#endif
+};
+
+/**
+ * @brief   Get the current time from a clock
+ *
+ * @param[in]   clock          ztimer clock to operate on
```

Comment:
Bit of a nitpick, but I think it would be more clear if the params call it a `ztimer64 clock`.
```suggestion
 * @param[in]   clock          ztimer64 clock to operate on
```

Reply:
done